### DISCUSSION TITLE
Tuya 6 channel scene switch

### DISCUSSION THREAD

**Discussion by DHM47** [Koenkk/zigbee2mqtt#10323] [Devices]
I just got this Tuya scene switch branded by Zemismart, unfortunately it is not currently supported by zigbee2mqtt.
I got it mostly working, but I wanted to make sure that I did it correctly and I'm seeking support with the battery reading.

Converter code:  

const fzLocal = {
    zemismart_button_clicks: {
        cluster: 'manuSpecificTuya',
        type: ['commandGetData', 'commandSetDataResponse'],
        convert: (model, msg, publish, options, meta) => {
            const button = msg.data.dp; 
            const actionValue = tuya.getDataValue(msg.data.datatype, msg.data.data);
            const lookup = {
                0: 'single',
                1: 'double',
                2: 'hold',
            };
            const action = lookup[actionValue];
            return {
                action: `button_${button}_${action}`,
            };
        },
    },
};
const definition = {
    fingerprint: [
            {modelID: 'TS0601', manufacturerName: '_TZE200_zqtiam4u'},
            ],
    model: 'ZM-RM02',
    vendor: 'Zemismart',
    description: 'Smart 6 key scene switch (Portable and Battery powered)',
    fromZigbee: [
        fzLocal.zemismart_button_clicks,
        fz.battery,
    ],
    toZigbee: [],
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genBasic']);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
        await reporting.batteryPercentageRemaining(endpoint);
    },
    exposes: [e.battery(), e.action([
        'button_1_hold', 'button_1_single', 'button_1_double',
        'button_2_hold', 'button_2_single', 'button_2_double',
        'button_3_hold', 'button_3_single', 'button_3_double',
        'button_4_hold', 'button_4_single', 'button_4_double',
        'button_5_hold', 'button_5_single', 'button_5_double',
        'button_6_hold', 'button_6_single', 'button_6_double'])],
};

![image](https://user-images.githubusercontent.com/6565963/147239072-cfa38869-f9fe-4f9e-9454-8dd225a1d79f.png)

**Comment by Koenkk** [comment]
Can you share the data/database.db entry of this device?

**Reply by DHM47** [reply]
{"id":22,"type":"EndDevice","ieeeAddr":"0x0c4314fffe5cbc52","nwkAddr":27315,"manufId":4098,"manufName":"_TZE200_zqtiam4u","powerSource":"Battery","modelId":"TS0601","epList":[1],"endpoints":{"1":{"profId":260,"epId":1,"devId":81,"inClusterList":[0,4,5,61184],"outClusterList":[25,10],"clusters":{"genBasic":{"attributes":{"65503":"\u0000\u0000\u0000\u0000\u0005\u0000\u0000\u0000\u0000\u0011z\u001bN)\u0012{\u001bN)\u0012\u0000\u0000\u0000\u0000\u0005\u0000\u0000\u0000\u0000\u0011\u0000\u0000\u0000\u0000\u0005\u0000\u0000\u0000\u0000\u0011","65506":31,"65508":1,"modelId":"TS0601","manufacturerName":"_TZE200_zqtiam4u","powerSource":3,"zclVersion":3,"stackVersion":0,"hwVersion":1,"dateCode":"","appVersion":68}}},"binds":[{"cluster":0,"type":"endpoint","deviceIeeeAddress":"0x00124b0024c0ef2e","endpointID":1},{"cluster":1,"type":"endpoint","deviceIeeeAddress":"0x00124b0024c0ef2e","endpointID":1}],"configuredReportings":[],"meta":{}}},"appVersion":68,"stackVersion":0,"hwVersion":1,"dateCode":"","zclVersion":3,"interviewCompleted":true,"meta":{},"lastSeen":1640300517993,"useImplicitCheckin":true}

**Reply by Koenkk** [reply]
Regarding battery, the device doesn't support the standard cluster that reports this value (`1` should be in `inClusterList` for that), so either:
- Device doesn't support battery reporting
- Battery % is reported via the TuYa manufacturer specific cluster (`61184`). You can try to find the data point via https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html#_2-signin-up-for-tuya-iot-platform

**Reply by DHM47** [reply]
Unfortunately I do not have a Tuya Zigbee gateway, also battery level is not that crucial for such devices. (General note: it accepts two 2032 batteries working in parallel, so I don't think it will run out of power anytime soon :) ) 

So is everything else good? If so, I'll update both the main repository (for the docs) and the herdsman repo (for the converter).

**Reply by Koenkk** [reply]
yes, please make a pr

**Reply by DavidFFerreira** [reply]
@DHM47 I have the device, the hub and the sniffer.....do you want to help me to find out?

**Reply by DHM47** [reply]
Sure

**Comment by DHM47** [comment]
@Koenkk with the latest update the scene switch stopped working, even my external converter doesn't seem to be recognized.

This what I get in the log:
 No converter available for 'ZM-RM02' with cluster 'manuSpecificTuya' and type 'commandDataResponse' and data '{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":5,"fn":0,"seq":1536}'

**Reply by Koenkk** [reply]
Let's continue in #10615

**Comment by DHM47** [comment]
@DavidFFerreira see issue #10615. @RJArmitage has solved the battery issue.
